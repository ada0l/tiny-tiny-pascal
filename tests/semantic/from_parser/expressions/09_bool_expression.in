var a: Double = 10;
    b: Double = 11;
    c: Double = 12;
    d: Double = 13;

begin
    writeln(2 <> 3);
    writeln(2 < 3);
    writeln(2 <= 3);
    writeln(2 >= 3);
    writeln(2 > 3);
    writeln(2 = 3);

    writeln(2 or 3);
    writeln(2 or 3 or 4 or 5 or 6);
    writeln(2 and 3 or 4 and 5);
    writeln(not 2 and not 3 or not 4 and not 5);
end.
