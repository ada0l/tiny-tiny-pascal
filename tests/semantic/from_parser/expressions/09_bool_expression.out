main
   program name:
      without name
   decls:
      vars:
         var
            a 
            type:
               Double 
            10
         var
            b 
            type:
               Double 
            11
         var
            c 
            type:
               Double 
            12
         var
            d 
            type:
               Double 
            13
   block:
      compounds statements
         writeln 
            <>
               2
               3
         writeln 
            <
               2
               3
         writeln 
            <=
               2
               3
         writeln 
            >=
               2
               3
         writeln 
            >
               2
               3
         writeln 
            =
               2
               3
         writeln 
            or
               2
               3
         writeln 
            or
               or
                  or
                     or
                        2
                        3
                     4
                  5
               6
         writeln 
            or
               and
                  2
                  3
               and
                  4
                  5
         writeln 
            or
               and
                  not
                     2
                  not
                     3
               and
                  not
                     4
                  not
                     5
------------------------------------------------------------------------------------------------------------------------------------------------------
| id                          | type of object              | ret type                                                                               |
------------------------------------------------------------------------------------------------------------------------------------------------------
| boolean                     | type                        | boolean                                                                                |
| integer                     | type                        | integer                                                                                |
| double                      | type                        | double                                                                                 |
| char                        | type                        | char                                                                                   |
| string                      | type                        | string                                                                                 |
| writeln                     | procedure                   | writeln                                                                                |
| write                       | procedure                   | write                                                                                  |
| read                        | procedure                   | read                                                                                   |
| random                      | function                    | double                                                                                 |
======================================================================================================================================================
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
| sqrt                        | function                    | double                                                                                 |
======================================================================================================================================================
| x                           | param                       | double                                                                                 |
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------------------------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------------------------------------------------------------
| id                          | type of object              | ret type                                                                               |
------------------------------------------------------------------------------------------------------------------------------------------------------
| a                           | var                         | double                                                                                 |
| b                           | var                         | double                                                                                 |
| c                           | var                         | double                                                                                 |
| d                           | var                         | double                                                                                 |
------------------------------------------------------------------------------------------------------------------------------------------------------

